sdk/Lusid.Sdk.Tests/tutorials/Instruments/CreditDefaultSwapCashFlow.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Lusid.Sdk.Api;
+using Lusid.Sdk.Model;
+using Lusid.Sdk.Tests.Tutorials.Ibor;
+using Lusid.Sdk.Tests.Utilities;
+using Lusid.Sdk.Utilities;
+using NUnit.Framework;
+
+namespace Lusid.Sdk.Tests.tutorials.Instruments
+{
+    public class CreditDefaultSwapCashFlow
+    {
+        private ILusidApiFactory _apiFactory;
+        private TestDataUtilities _testDataUtilities;
+        private ITransactionPortfoliosApi _transactionPortfoliosApi;
+        private IPortfoliosApi _portfoliosApi;
+        private static IConfigurationRecipeApi _recipeApi;
+        private IAggregationApi _aggregationApi;
+        
+        private static readonly string ValuationDateKey = "Analytic/default/ValuationDate";
+        private static readonly string ValuationPv = "Valuation/PV/Amount";
+        private static readonly string ValuationCcy = "Valuation/PV/Ccy";
+        private static readonly string Luid = "Instrument/default/LusidInstrumentId";
+        private static readonly string InstrumentName = "Instrument/default/Name";
+        private static readonly string InstrumentTag = "Analytic/default/InstrumentTag";
+        
+        private static readonly List<AggregateSpec> ValuationSpec = new List<AggregateSpec>
+        {
+            new AggregateSpec(ValuationDateKey, AggregateSpec.OpEnum.Value),
+            new AggregateSpec(ValuationPv, AggregateSpec.OpEnum.Value),
+            new AggregateSpec(ValuationCcy, AggregateSpec.OpEnum.Value),
+            new AggregateSpec(InstrumentName, AggregateSpec.OpEnum.Value),
+            new AggregateSpec(InstrumentTag, AggregateSpec.OpEnum.Value),
+            new AggregateSpec(Luid, AggregateSpec.OpEnum.Value)
+        };
+
+        [OneTimeSetUp]
+        public void SetUp()
+        {
+            _apiFactory = TestLusidApiFactoryBuilder.CreateApiFactory("secrets.json");
+            _transactionPortfoliosApi = _apiFactory.Api<ITransactionPortfoliosApi>();
+            _portfoliosApi = _apiFactory.Api<IPortfoliosApi>();
+            _recipeApi = _apiFactory.Api<IConfigurationRecipeApi>();
+            _aggregationApi = _apiFactory.Api<IAggregationApi>();
+            _testDataUtilities = new TestDataUtilities(
+                _apiFactory.Api<ITransactionPortfoliosApi>(),
+                _apiFactory.Api<IInstrumentsApi>(),
+                _apiFactory.Api<IQuotesApi>(),
+                _apiFactory.Api<IComplexMarketDataApi>());
+        }
+        
+        [Test]
+        public void ExamplePortfolioCashFlowsForCreditDefaultSwaps()
+        {
+            // CREATE portfolio
+            var portfolioScope = Guid.NewGuid().ToString();
+            var portfolioId = _testDataUtilities.CreateTransactionPortfolio(portfolioScope);
+
+            // CREATE CDS
+            var cds = InstrumentExamples.CreateExampleCreditDefaultSwap() as CreditDefaultSwap;
```

Comment:
Curious here, and might be a C-sharpism I'm missing, but what does the `as CreditDefaultSwap` do here and is it needed? Does it validate the type or?

Reply:
It casts the object into the type `CDS`. The CDS's details are used later when upserting market data rather than hardcoding "SNR" for seniority etc